app.py

```diff
@@ -144,7 +167,10 @@ def render_and_submit_path(self, path, fields, first_frame, last_frame, sg_publi
                                       first_frame, last_frame,
                                       fields.get("version", 0), 
                                       fields.get("name", "Unnamed"),
-                                      color_space)
+                                      color_space, replace_data)
+
+        sub_first_frame = replace_data["sg_head_in"] if replace_data["sg_head_in"] else first_frame
```

Comment:
Since we are always rendering head_in to tail_out, this should be unnecessary... `first_frame == sg_head_in && last_frame == sg_tail_out` always in our pipeline. The change should be made in the code calling tk-multi-reviewsubmission to always pass in head_in and tail_out. tk-multi-reviewsubmission should be agnostic.

Reply:
In case we have an asset or sequence or project level file, head_in, tail_out will be None whereas first_frame, last_frame will be populated based on the files we are trying to publish (i.e frames available on disk) This will make sure the frame count and frame range calculations in submitter do not give errors.

Reply:
Let's talk about this at the touchbase. I still don't quite understand.

Reply:
Sorry, I think I didn't pay attention to your statement properly. For LEMONY3 atleast, we are not rendering head_in to tail_out, but only cut_in to cut_out. The head_in to head_out consists of just slate and tail_in to tail_out has black frames, being added in the review submission movie creation nuke graph.

Reply:
We should always be rendering head_in to tail_out, even if its only rendering black frames / slate.

Reply:
Okay, when I say rendered frames, I mean the frames rendered by the artist. Which will be his working range. 
If we are adding extra slate or black frames to the movie, that is a setup we are doing in the review submission nuke script. 
To ask artists to render and keep them doesn't make sense. We would be rendering a lot of extra frames, copying them during publish etc. wasting render time & resources and disk space.